### PR TITLE
Cleanup metrics in ratelimit middleware

### LINKED ISSUE #773: Deduplicate metrics
In many (all?) places, we are using two separate metrics to count failed/successful states. E.g. we have `code_issued_count` and `code_issue_error_count` [here](https://github.com/google/exposure-notifications-verification-server/blob/44a426a524442aedc164f3f2a7df809adb55c3cf/pkg/controller/issueapi/metrics.go#L60-L72).

Having two metrics makes it impossible to calculate an error ratio which is a very important metric to monitor. And we will need to add more metrics if we want to further break down what caused the error.

Instead we should have a single `code_issued_count` metric, with additional labels like `error_reason` (short text on why it failed) and `blame` (whether the failure is to blame the client or the server). This, together with the realm id information that we already have, allows us to setup flexible alerting rules:

- Overall `blame=server` error rate/ratio increase
- Per-realm-id `blame=client` error rate/ratio increase
- One particular `error_reason` rate/ratio increase

### LINKED ISSUE #906: Build tag missing from some metrics
### TL;DR
All metrics should have a label with the build version, it seems to have disappeared in some cases. Let's find out why.

**Expected behavior**

A label for build_id and build_version on all metrics.

**Observed behavior**

Both labels were missing in the /rate_limited.* metrics

### PR DESCRIPTION
**Pull request by yegle**
Fixes #773

Also appear to fix #906

### CODE REVIEW

File: pkg/ratelimit/limitware/metrics.go [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+)
+
+const metricPrefix = observability.MetricRoot + "/ratelimit/limitware"
+
+var (
+	mRequest = stats.Int64(metricPrefix+"/request", "requests seen by middleware", stats.UnitDimensionless)
+)
+
+func init() {
+	enobservability.CollectViews(append(redis.ObservabilityMetricViews,
```
Comment (icco) [inline pkg/ratelimit/limitware/metrics.go:32]: Should we panic here if this fails?

File: pkg/ratelimit/limitware/metrics.go [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+// limitations under the License.
+
+package limitware
+
+import (
+	enobservability "github.com/google/exposure-notifications-server/pkg/observability"
+	"github.com/google/exposure-notifications-verification-server/pkg/observability"
+	"github.com/opencensus-integrations/redigo/redis"
+	"go.opencensus.io/stats"
+	"go.opencensus.io/stats/view"
```
Comment (icco) [inline pkg/ratelimit/limitware/metrics.go:22]: We usually seperate this with newlines into three blocks env imports, en imports and third party imports

File: pkg/ratelimit/limitware/middleware.go [hunk 3]
```diff
@@ -84,58 +73,10 @@ func NewMiddleware(ctx context.Context, s limiter.Store, f httplimit.KeyFunc, op
...
-	if err := view.Register(&view.View{
-		Name:        MetricPrefix + "/rate_limited_count",
-		Measure:     rl,
-		Aggregation: view.Count(),
-		TagKeys:     []tag.Key{},
-	}); err != nil {
-		return nil, fmt.Errorf("stat view registration failure: %w", err)
-	}
-
-	if err := view.Register(redis.ObservabilityMetricViews...); err != nil {
```
Comment (icco) [inline pkg/ratelimit/limitware/middleware.go:127]: Are these registered elswhere?
